kubeconfig_plugin.py

```diff
@@ -117,57 +138,106 @@ class ErrorOutput:
-        except Exception:
-            print(
-                "client certificate and key data missing, trying to extract"
-                " token"
-            )
+            current_context = kubeconfig["current-context"]
+        except KeyError:
+            return "error", ErrorOutput(
+                "The provided kubeconfig file does not have a current-context set. Please set a current context to use.")
 
-        if (
-            "client_key_data" not in output
-            and "client_certificate_data" not in output
-        ):
-            try:
-                output["token"] = kcl._user.value["token"]
-            except Exception:
-                print("token missing for user in kubeconfig")
-
-        if (
-            "client_key_data" not in output
-            and "client_certificate_data" not in output
-            and "token" not in output
-        ):
+        try:
+            context = None
+            for ctx in kubeconfig["contexts"]:
+                if ctx["name"] == current_context:
+                    context = ctx["context"]
+            if context is None:
+                return "error", ErrorOutput(
+                    "Failed to find a context named {} in the kubeconfig file.".format(current_context))
+            current_cluster = context["cluster"]
+            current_user = context["user"]
+        except Exception as e:
             return "error", ErrorOutput(
-                1, "Both client data and token missing, exiting!"
-            )
+                "Failed to find a context named {} in the kubeconfig file: {}".format(current_context, e.__str__()))
 
-        print(output)
+        try:
+            cluster = None
+            for cl in kubeconfig["clusters"]:
+                if cl["name"] == current_cluster:
+                    cluster = cl["cluster"]
+            if cluster is None:
+                return "error", ErrorOutput(
+                    "Failed to find a cluster named {} in the kubeconfig file.".format(current_cluster))
+        except Exception as e:
+            return "error", ErrorOutput(
+                "Failed to find a cluster named {} in the kubeconfig file: {}".format(current_cluster, e.__str__()))
+
+        try:
+            user = None
+            for u in kubeconfig["users"]:
+                if u["name"] == current_user:
+                    user = u["user"]
+            if user is None:
+                return "error", ErrorOutput(
+                    "Failed to find a user named {} in the kubeconfig file.".format(current_user))
+        except Exception as e:
+            return "error", ErrorOutput(
+                "Failed to find a user named {} in the kubeconfig file: {}".format(current_user, e.__str__()))
 
-        return "success", kubeconfig_output_schema.unserialize(output)
-    except Exception:
-        return "error", ErrorOutput(1, "Failure in parsing kubeconfig:")
+        output = SuccessOutput(
+            Connection(cluster["server"]),
```

Comment:
can we first assign cluster["server"] as host and then use that, it'll reduce confusion in the future